cli/pcluster/configure/networking.py

```diff
@@ -0,0 +1,264 @@
+
+    @abc.abstractmethod
+    def _create(self, aws_region_name, vpc_id, vpc_cidr, subnet_cidrs, internet_gateway_id, compute_subnet_size):
+        pass
+
+    @staticmethod
+    def _get_cfn_parameters(aws_region_name, vpc_id, internet_gateway_id):
+        availability_zone = _get_availability_zone(aws_region_name)
+        return [
+            {"ParameterKey": "AvailabilityZone", "ParameterValue": availability_zone},
+            {"ParameterKey": "InternetGatewayId", "ParameterValue": internet_gateway_id},
+            {"ParameterKey": "VpcId", "ParameterValue": vpc_id},
+        ]
+
+
+class PublicNetworkConfig(BaseNetworkConfig):
+    """The public configuration that creates one public subnet with master and compute fleet."""
+
+    def __init__(self):
+        super(PublicNetworkConfig, self).__init__(
+            config_type="Master and compute fleet in the same public subnet",
+            template_name="public",
+            stack_name_prefix="pub"
+        )
+
+    def get_cfn_parameters(self, aws_region_name, vpc_id, internet_gateway_id, public_cidr):
+        """Create cloudformation-compatible stack parameter given the variables."""
+        parameters = super(PublicNetworkConfig, self)._get_cfn_parameters(aws_region_name, vpc_id, internet_gateway_id)
+        parameters.append({"ParameterKey": "PublicCIDR", "ParameterValue": public_cidr})
+        return parameters
+
+    def _create(self, aws_region_name, vpc_id, vpc_cidr, subnet_cidrs, internet_gateway_id, compute_subnet_size):  # noqa D102
+        public_cidr = get_subnet_cidr(
+            vpc_cidr=vpc_cidr, occupied_cidr=subnet_cidrs, min_subnet_size=compute_subnet_size + MASTER_SUBNET_IPS
+        )
+        _validate_cidr(public_cidr)
+        parameters = self.get_cfn_parameters(aws_region_name, vpc_id, internet_gateway_id, public_cidr)
+        stack_output = _create_network_stack(aws_region_name, self, parameters)
+        return {"master_subnet_id": get_stack_output_value(stack_output, "PublicSubnetId"), "use_public_ips": "true"}
+
+
+class PublicPrivateNetworkConfig(BaseNetworkConfig):
+    """The publicprivate configuration that creates one public subnet for master and one private subnet for compute."""
+
+    def __init__(self):
+        super(PublicPrivateNetworkConfig, self).__init__(
+            config_type="Master in a public subnet and compute fleet in a private subnet",
+            template_name="public-private",
+            stack_name_prefix="pubpriv"
+        )
+
+    def get_cfn_parameters(self, aws_region_name, vpc_id, internet_gateway_id, public_cidr, private_cidr):
+        """Create cloudformation-compatible stack parameter given the variables."""
+        parameters = super(PublicPrivateNetworkConfig, self)._get_cfn_parameters(
+            aws_region_name, vpc_id, internet_gateway_id
+        )
+        parameters.append({"ParameterKey": "PublicCIDR", "ParameterValue": public_cidr})
```

Comment:
minor:
we could create a `_build_cfn_param(key, value)` that automatically create the json item:
`return {"ParameterKey": key, "ParameterValue": value}`

it's up to you.